src/super_context.ml

```diff
@@ -261,6 +177,34 @@ end = struct
         Exn.code_error "Super_context.Env.get called on invalid directory"
           [ "dir", Path.to_sexp dir ]
 
+  let external_ t ~dir =
+    let rec loop t node =
```

Comment:
Just a note: if we wrote this function this way instead:

```ocaml
let external_ =
  let rec loop t node =
    ...
  in
  fun t ~dir ->
     ...
```

Then we are sure to not accidentally use the wrong `~dir`  inside `loop`.